sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -325,3 +325,51 @@ TEST(Context, KeyComparison)
   EXPECT_NE(localKey1Copy, localKey2);
   EXPECT_NE(localKey2Copy, localKey1);
 }
+
+TEST(Context, Deadline)
+{
+  auto const deadline = Azure::DateTime(2021, 4, 1, 23, 45, 15);
+  Context::Key const key1;
+  Context::Key const key2;
+  
+  {
+    Context ctx;
+    EXPECT_FALSE(ctx.HasDeadline());
+    EXPECT_EQ(ctx.GetDeadline(), Azure::DateTime::max());
+
+    ctx.Cancel();
+    EXPECT_TRUE(ctx.HasDeadline());
+    EXPECT_EQ(ctx.GetDeadline(), Azure::DateTime::min());
```

Comment:
@JeffreyRichter It is odd that we lose information such as the actual deadline time after `ctx.Cancel()` is called. Should we set the deadline to `::now()` when `Cancel()` get's called? This way the caller knows when the context was cancelled.

Reply:
I don't think it will matter in real life. So, setting it to now instead of min is fine with me.

Reply:
@ahsonkhan , let's track that feature request separately, please open an issue if you think we should preserve the cancelation time. We can store `system_clock::now()` as cancelation but then the context state should have `bool isCanceled`. Currently, all that `Cancel()` does is it sets the deadline to `DateTime::min()`. What I don't want to happen is the context getting resurrected once system clock gets adjusted.

Reply:
>  let's track that feature request separately, please open an issue

https://github.com/Azure/azure-sdk-for-cpp/issues/2038